src/backend/optimizer/path/allpaths.c

```diff
@@ -235,6 +247,23 @@ static void
 set_base_rel_sizes(PlannerInfo *root)
 {
 	Index		rti;
+	int			resultRelIndex = root->parse->resultRelation;
+	RelOptInfo *resultRel;
+
+	/*
+	 * If result relation is excluded by constraint exclusion we shouldn't
+	 * plan this sub-query
```

Comment:
```suggestion
	 * If result relation is excluded by constraint exclusion we shouldn't
	 * plan this sub-query. Result relation is set to zero for SELECT
	 * queries. simple_rel_array and simple_rte_array are filled by 
	 * zeroes during allocation and never use zero-indexed item.
	 * So, SELECT queries bypass this logic safely.
```

@maksm90, Is this enough description for this moment mentioned by yours?

Reply:
Sure